### PR TITLE
Fixed the API Integration issue and Folder Structures

### PR DESCRIPTION
**Pull request by adars-kafle**

### CODE REVIEW

File: src/context/AuthContext.tsx [hunk 1]
```diff
@@ -1,28 +1,95 @@
...
+      const response = await authApi.getCurrentUser(); // send api request to get the current user
+      if (response) {
+        setUser(response);
+        localStorage.setItem("user", JSON.stringify(response)); // store the user info in local storage
+      } else {
+        setUser(null);
+        localStorage.removeItem("user");
+      }
+    } catch (error) {
+      console.log("Error fetching user: ", error);
```
Comment (sudipsth08) [inline src/context/AuthContext.tsx:36]: create a util that logs only in development

File: src/hooks/useSupplier.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query";
+import { supplierApi } from "../services/supplierServices";
+import { type SupplierFormInputs } from "../interfaces/types";
+
+export const useSuppliers = (skip: number = 0, limit: number = 10) => {
+  return useQuery({
+    queryKey: ["suppliers", skip, limit], // this array uniquely identifies the query.
```
Comment (sudipsth08) [inline src/hooks/useSupplier.ts:7]: creeate a const file for queryKey

**Review by sudipsth08 — CHANGES_REQUESTED** [review]

File: src/hooks/useSupplier.ts (hunk 2 above)
Comment (adars-kafle) [reply src/hooks/useSupplier.ts]: I added it as per your request.

File: src/context/AuthContext.tsx (hunk 1 above)
Comment (adars-kafle) [reply src/context/AuthContext.tsx]: I added the logger dai, it works only in the dev environment.

**Pull request merged**